### PR TITLE
[admission-policy-engine] Add mutations

### PR DESCRIPTION
**Pull request by yalosev**
## Description
Add mutations support

## Why do we need it, and what problem does it solve?
It's a common functionality of the gatekeeper. We didn't want to have a mutationg webhook with wildcard resources. But now we are calculation resources from the CRD's and handle them.

## Checklist
- [x] The code is covered by unit tests.
- [ ] e2e tests passed.
- [x] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: admission-policy-engine
type: feature
summary: Add gatekeeper [mutations](https://open-policy-agent.github.io/gatekeeper/website/docs/mutation) support
```

### CODE REVIEW

File: modules/015-admission-policy-engine/docs/README.md [hunk 1]
```diff
@@ -64,3 +64,32 @@ spec:
 
 To apply the policy, it will be sufficient to set the label `operation-policy.deckhouse.io/enabled: "true"` on the desired namespace.
 The above policy is generic and recommended by Deckhouse team. Similarly, you can configure your own policy with the necessary settings.
+
+### Modifying kubernetes resources
+
+The module also allows you to use the Gatekeeper's CustomResources, to easily modify objects in the cluster, such as
+- `AssignMetadata` - defines changes to the metadata section of a resource
+- `Assign` - any change outside the metadata section
+- `ModifySet` - adds or removes entries from a list, such as the arguments to a container
```
Comment (z9r5) [inline modules/015-admission-policy-engine/docs/README.md:73]:
```suggestion
### Modifying Kubernetes resources

The module also allows you to use the Gatekeeper's Custom Resources to easily modify objects in the cluster, such as
- `AssignMetadata` — defines changes to the metadata section of a resource.
- `Assign` —  any change outside the metadata section.
- `ModifySet` —  adds or removes entries from a list, such as the arguments to a container.
```

File: modules/015-admission-policy-engine/docs/README_RU.md [hunk 2]
```diff
@@ -65,3 +65,32 @@ spec:
 ```
 
 Для применения приведенной политики достаточно навесить лейбл `operation-policy.deckhouse.io/enabled: "true"` на желаемый namespace. Политика, приведенная в примере, рекомендована для использования командой Deckhouse. Аналогичным образом вы можете создать собственную политику с необходимыми настройками.
+
+### Изменение ресурсов kubernetes
+
+Модуль также позволяет использовать CRD Gatekeeper, для легкой модификации объектов в кластере, такие как:
+- `AssignMetadata` - для изменения секции metadata в ресурсе
+- `Assign` - изменения других полей, кроме metadata
+- `ModifySet` - добавление или удаление значений из списка, например аргументы для контейнера
```
Comment (z9r5) [inline modules/015-admission-policy-engine/docs/README_RU.md:74]:
```suggestion
### Изменение ресурсов Kubernetes

Модуль также позволяет использовать custom resource'ы Gatekeeper, для легкой модификации объектов в кластере, такие как:
- `AssignMetadata` — для изменения секции metadata в ресурсе.
- `Assign` — изменения других полей, кроме metadata.
- `ModifySet` — добавление или удаление значений из списка, например аргументы для контейнера.
```

File: modules/015-admission-policy-engine/docs/README_RU.md [hunk 3]
```diff
@@ -65,3 +65,32 @@ spec:
 ```
 
 Для применения приведенной политики достаточно навесить лейбл `operation-policy.deckhouse.io/enabled: "true"` на желаемый namespace. Политика, приведенная в примере, рекомендована для использования командой Deckhouse. Аналогичным образом вы можете создать собственную политику с необходимыми настройками.
+
+### Изменение ресурсов Kubernetes
+
+Модуль также позволяет использовать custom resource'ы Gatekeeper, для легкой модификации объектов в кластере, такие как:
+- `AssignMetadata` — для изменения секции metadata в ресурсе.
+- `Assign` — изменения других полей, кроме metadata.
+- `ModifySet` — добавление или удаление значений из списка, например аргументы для контейнера.
```
Comment (z9r5) [inline modules/015-admission-policy-engine/docs/README_RU.md:74]:
```suggestion
- `AssignMetadata` — для изменения секции metadata в ресурсе.
- `Assign` — для изменения других полей, кроме metadata.
- `ModifySet` — для добавления или удаления значений из списка, например аргументов для запуска контейнера.
```

File: modules/015-admission-policy-engine/openapi/values.yaml [hunk 4]
```diff
@@ -36,6 +36,20 @@ properties:
               type: array
               items:
                 type: string
+      trackedMutateResources:
+        type: array
+        default: [ ]
```
Comment (nabokihms) [inline modules/015-admission-policy-engine/openapi/values.yaml:41]: NIT
```suggestion
        default: []
```

**Review by nabokihms — APPROVED** [review]

**Pull request merged**